### PR TITLE
Add more image data for exercise images

### LINKED ISSUE #931: More image data when submitting an exercise
### Use case

At the moment, when uploading an image while submitting an exercise, we only upload the image itself. We should expand on this and add other necessary fields, like we already do in the web version. Also important is adding a message about the license of the images:

### Proposal

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by Oknarb258**
# Proposed Changes
- Added license metadata collection for exercise images to comply with CC BY-SA 4.0
- Modified `AddExerciseProvider` to store license fields (title, author, URLs, style) per image
- Created `ImageDetailsForm` widget for collecting metadata before adding images
- Updated `Step5Images` to show metadata form instead of direct upload
- Added comprehensive unit tests covering edge cases and metadata handling

## Related Issue(s)

If applicable, please link to any related issues (`Closes #931`,
More image data when submitting an exercise #931

## Please check that the PR fulfills these requirements

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Set a 100 character limit in your editor/IDE to avoid white space diffs in the PR
  (run `dart format .`)
- [x] Updated/added relevant documentation (doc comments with `///`).
- [ ] Added relevant reviewers.

### CODE REVIEW

File: lib/providers/add_exercise.dart [hunk 1]
```diff
@@ -21,6 +21,10 @@ class AddExerciseProvider with ChangeNotifier {
 
   List<File> get exerciseImages => [..._exerciseImages];
   List<File> _exerciseImages = [];
+
+  // Map storing image metadata (license info, style), keyed by file path
+  final Map<String, Map<String, String>> _imageDetails = {};
```
[the remark this answers is not in the document]
Comment (rolandgeider) [reply lib/providers/add_exercise.dart:27]: oh sure, I wasn't saying that it didn't work or so

**Comment by rolandgeider** [comment]
🚀 

(I just recreated the generated files to fix the test btw)
